src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs

```diff
@@ -363,13 +391,20 @@ public void StopProcessingNextRequest(bool serverInitiated)
                     stream.Abort(new IOException(CoreStrings.Http2StreamAborted, connectionError));
                 }
 
-                // Use the server _serverActiveStreamCount to drain all requests on the server side.
-                // Can't use _clientActiveStreamCount now as we now decrement that count earlier/
-                // Can't use _streams.Count as we wait for RST/END_STREAM before removing the stream from the dictionary
-                while (_serverActiveStreamCount > 0)
+                // For some reason, this loop doesn't terminate when we're trying to abort.
+                // Since we're making a narrow fix for a patch, we'll bypass it in such scenarios.
+                // TODO: This is probably a bug - something in here should probably detect aborted
```

Comment:
Issue?

Reply:
Obviously, I couldn't file a public issue at the time.